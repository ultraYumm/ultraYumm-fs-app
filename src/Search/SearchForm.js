import React, { Component } from "react";
import "../Font/Font.css";
import config from "../config";
import GoButton from "../FormElements/GoButton";
import "../FormElements/FormElements.css";
import { withRouter } from "react-router-dom";
import { Auth } from 'aws-amplify';



class SearchForm extends Component {

  componentDidMount () { Auth.currentAuthenticatedUser().then(user => {

    let id = user.attributes.sub
    let username = user.username

    this.props.getUser(id, username)
   
  });

    }


  render() {
    const onSubmitForm = (e) => {
      e.preventDefault();

      let searchTerm = e.target.searchTerm.value;
      this.props.handleUpdate(searchTerm);
      this.props.routerProps.history.push("/search-results");
      const apId = config.NUTRI_APID;
      const apiKey = config.API_NUTRI_KEY;
      const remoteUser = config.NUTRI_REMOTEUSER;
      const detailed = config.NUTRI_DETAILED;
      const getURL = config.API_NUTRI_ENDPOINT;

      function formatQueryParams(params) {
        const queryItems = Object.keys(params).map(
          (key) =>
            `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`
        );
        return queryItems.join("&");
      }

      const displayResults = (responseJson) => {
        let searchResults = responseJson;
        return this.props.handleResults(searchResults);
      };

      function getResults(query) {
        const params = {
          query: searchTerm,
          detailed: detailed,
        };

        const queryString = formatQueryParams(params);
        const url = getURL + "?" + queryString;

        const options = {
          headers: new Headers({
            "x-app-id": apId,
            "x-app-key": apiKey,
            "x-remote-user-id": remoteUser,
          }),
        };

        fetch(url, options)
          .then((response) => {
            if (response.ok) {
              return response.json();
            }
            throw new Error(response.statusText);
          })
          .then((responseJson) => displayResults(responseJson))
          .catch((err) => {});
      }
      getResults(searchTerm)
    };

    return (
      
        <form className= "myPlans" onSubmit={onSubmitForm}>
          <h2 className="white"><i className ="fas fa-skiing"></i>quick search!</h2>
          <label htmlFor="searchTerm" className="white montebello searchTerm"><i className ="fas fa-cookie-bite"></i> By name or brand 
          </label>
          <input
            className="skinBackground purple"
            name="searchTerm"
            type="text" 
            placeholder="Granola"
            required
            id="searchTerm"
          />

          <GoButton
           username = {this.props.username}/>
        </form>
  
    );
  }
}

export default withRouter(SearchForm);
