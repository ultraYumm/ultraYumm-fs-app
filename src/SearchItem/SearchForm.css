
 .uYtitle {
    width:20%;
    display: block;
    padding-top: 30px;
    margin-left: auto;
    margin-right: auto;
  }


.hero {
  font-size: 20px;
  text-align: center;
  font-weight: lighter;
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: -15px;
}
