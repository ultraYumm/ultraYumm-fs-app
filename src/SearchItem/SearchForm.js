import React, { Component } from 'react';
import './SearchForm.css';
import '../Font/Font.css';
import uYtitle from '../Images/uYtitle.png';
import SearchBar from './SearchBar';


class SearchForm extends Component {
  render() {
    return (

    <div >
        
        <img className= "uYtitle" src={uYtitle} alt= "ultraYumm logo"/>
       
        <div  className= "hero black montebello">Get more energy!<span>&nbsp;&nbsp;</span><span class= "lighter">pack lighter</span></div>
       
       
        <SearchBar></SearchBar>
   
    </div>
   
   );
  }
}
export default SearchForm;