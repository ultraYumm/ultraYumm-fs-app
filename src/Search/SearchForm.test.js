import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <BrowserRouter>
     <SearchForm />
  </BrowserRouter> 
  , div);
  ReactDOM.unmountComponentAtNode(div);
});


it('renders the UI as expected', () => {
  const tree = renderer
    .create( <BrowserRouter>
       <SearchForm />
       </BrowserRouter> )
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
