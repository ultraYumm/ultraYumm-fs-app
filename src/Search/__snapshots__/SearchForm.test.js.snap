// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="white"
    >
      <i
        className="fas fa-skiing"
      />
       Quick search!
    </h2>
    <label
      className="white montebello searchTerm"
      htmlFor="searchTerm"
    >
      <i
        className="fas fa-cookie-bite"
      />
       By name or brand
    </label>
    <input
      className="skinBackground purple"
      id="searchTerm"
      name="searchTerm"
      placeholder="Granola"
      required={true}
      type="text"
    />
    <div
      className="goContainer go"
    >
      <button
        className="white montebello blackBackground go"
        type="submit"
      >
        Go!
      </button>
    </div>
  </form>
</div>
`;
